src/main/java/org/terracotta/build/plugins/buildinfo/BuildInfoExtension.java

```diff
@@ -1,25 +1,25 @@
 package org.terracotta.build.plugins.buildinfo;
 
 import org.gradle.api.provider.Property;
+import org.gradle.api.provider.Provider;
 
-import java.time.Instant;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
 
-public abstract class BuildInfoExtension implements BuildInfo {
-  @Override
-  public abstract Property<Boolean> getHasLocalChange();
+public interface BuildInfoExtension {
+  Property<Boolean> getHasLocalChange();
 
-  @Override
-  public abstract Property<String> getBranch();
+  Property<String> getBranch();
 
-  @Override
-  public abstract Property<String> getCommitHash();
+  Property<String> getCommitHash();
 
-  @Override
-  public abstract Property<String> getRevision();
+  Property<String> getRevision();
 
-  @Override
-  public abstract Property<StructuredVersion> getVersion();
+  Property<StructuredVersion> getVersion();
 
-  @Override
-  public abstract Property<Instant> getBuildTimestamp();
+  Property<ZonedDateTime> getBuildTimestamp();
```

Comment:
Binding the timezone in here makes the downstream consumption much cleaner:
```groovy
terracotta.build.timestamp=${timestamp.get().format("uuuu-MM-dd 'at' HH:mm:ss z")}
```

Reply:
but a zoned data time is not a timestamp... this is a localised date time that holds a zone information...

Reply:
ok ok ;-) I just do not expect such return type when I see a method called getTimestamp. Usually a timestamp means something else than a time.